source/read/change-streams.txt

```diff
@@ -0,0 +1,272 @@
+-------------------------------
+
+To modify the change stream output, you can pass a list of pipeline stages as a
+parameter to the ``watch()`` method. You can include the following stages in the
+list:
+
+- ``$addFields`` or ``$set``: Adds new fields to documents
+- ``$match``: Filters the documents
+- ``$project``: Projects a subset of the document fields
+- ``$replaceWith`` or ``$replaceRoot``: Replaces the input document with the
+  specified document
+- ``$redact``: Restricts the contents of the documents
+- ``$unset``: Removes fields from documents
+
+The {+driver-short+} provides the ``Aggregates`` class, which includes helper methods
+for building the preceding pipeline stages. To learn more about ``Aggregates``, see
+the `API documentation <{+api+}/org/mongodb/scala/model/Aggregates$.html>`__.
+
+The following example passes a pipeline to ``watch()`` that uses the ``Aggregates.filter()``
+method to build the ``$match`` stage. This instructs the ``watch()`` method to output
+events only when update operations occur:
+
+.. literalinclude:: /includes/read/change-streams.scala
+   :start-after: start-change-stream-pipeline
+   :end-before: end-change-stream-pipeline
+   :language: scala
+   :dedent:
+
+Modify watch() Behavior
+-----------------------
+
+You can modify the behavior of the ``watch()`` method by chaining
+methods provided by the ``ChangeStreamObservable`` class. The following
+table describes some of these methods:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``fullDocument()``
+     - | Specifies whether to show the full document after the change, rather
+         than showing only the changes made to the document. To learn more about
+         this option, see the :ref:`scala-change-stream-pre-post-image` section of this
+         guide.
+
+   * - ``fullDocumentBeforeChange()``
+     - | Specifies whether to show the full document as it was before the change, rather
+         than showing only the changes made to the document. To learn more about
+         this option, see :ref:`scala-change-stream-pre-post-image`.
+
+   * - ``comment()``
+     - | Attaches a comment to the operation.
+
+   * - ``startAtOperationTime()``
+     - | Instructs the change stream to only provide changes that occurred at or after
+         the specified timestamp.
+
+   * - ``collation()``
+     - | Sets the collation to use for the change stream cursor.
+
+For a full list of ``watch()`` options, see `ChangeStreamObservable
+<{+api+}/org/mongodb/scala/ChangeStreamObservable.html>`__ in the API
+documentation.
+
+.. _scala-change-stream-pre-post-image:
+
+Include Pre-Images and Post-Images
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. important::
+
+   You can enable pre-images and post-images on collections only if your
+   deployment uses MongoDB v6.0 or later.
+
+By default, when you perform an operation on a collection, the
+corresponding change event includes only the delta of the fields
```

Comment:
S: some readers might not know "delta" in this context. could either reword or link to something like this:

https://en.wikipedia.org/wiki/Delta_encoding